File: templates/java/authentication/mutatingPostAuthentication.java
package io.fireboom.authentication;

import io.fireboom.plugins.AuthenticationHooks;
import io.fireboom.server.authentication.MutatingPostAuthenticationResponse;
import io.fireboom.server.hook.BaseRequestBodyWg;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class MutatingPostAuthentication extends AuthenticationHooks implements AuthenticationHooks.MutatingPostAuthentication {

    @Override
    public MutatingPostAuthenticationResponse execute(BaseRequestBodyWg body) {
        return new MutatingPostAuthenticationResponse(null, "ok", body.getUser());
    }
}

File: templates/golang/authentication/mutatingPostAuthentication.go
package authentication

import (
	"custom-go/pkg/base"
	"custom-go/pkg/plugins"
)

func MutatingPostAuthentication(hook *base.AuthenticationHookRequest) (*plugins.AuthenticationResponse, error) {
	return &plugins.AuthenticationResponse{User: hook.User, Status: "ok"}, nil
}
